### PR TITLE
Add withdrawAsEth to ERC20PositionWithdrawer

### PR DESCRIPTION
**Pull request by BrendanChou**
This is for positions with WETH as collateral (LETH, etc)

### CODE REVIEW

File: contracts/margin/external/ERC20/ERC20PositionWithdrawer.sol [hunk 1]
```diff
@@ -90,13 +90,13 @@ contract ERC20PositionWithdrawer is ReentrancyGuard
         returns (uint256, uint256)
     {
         // withdraw tokens
+        address withdrawnToken = ERC20Position(erc20Position).heldToken();
```
Comment (AntonioJuliano) [inline contracts/margin/external/ERC20/ERC20PositionWithdrawer.sol]: Was this change just for clarity?

File: contracts/margin/external/ERC20/ERC20PositionWithdrawer.sol [hunk 2]
```diff
@@ -120,4 +120,32 @@ contract ERC20PositionWithdrawer is ReentrancyGuard
...
+        // verify that WETH will be withdrawn
+        address token = ERC20Position(erc20Position).heldToken();
+        require(
+            token == WETH,
+            "ERC20PositionWithdrawer#withdrawAsEth: Withdrawn token must be WETH"
+        );
+
+        // withdraw tokens
+        uint256 amount = ERC20Position(erc20Position).withdraw(msg.sender);
+        assert(WETH9(token).balanceOf(address(this)) >= amount);
```
Comment (AntonioJuliano) [inline contracts/margin/external/ERC20/ERC20PositionWithdrawer.sol]: Not sure if this is necessary. It'll cost a reasonable amount of gas (external call + storage read), and the `WETH9(token).withdraw(amount);` will just fail if it doesn't have the balance anyways

File: contracts/margin/external/ERC20/ERC20PositionWithdrawer.sol [hunk 3]
```diff
@@ -120,4 +120,32 @@ contract ERC20PositionWithdrawer is ReentrancyGuard
...
+     * Withdraw WETH tokens from a position, unwrap the tokens, and send back to the trader
+     *
+     * @param  erc20Position  The address of the ERC20Position contract to withdraw from
+     * @return                The amount of ETH withdrawn
+     */
+    function withdrawAsEth(
+        address erc20Position
+    )
+        external
+        returns (uint256)
```
Comment (AntonioJuliano) [inline contracts/margin/external/ERC20/ERC20PositionWithdrawer.sol]: So `withdraw` is `nonReentrant`, should this be as well?

It seems like in general we're being a bit inconsistent with what we're making `nonReentrant`. In some places we seem to add it just to be safe even when we don't believe it's necessary, and others not

File: contracts/margin/external/ERC20/ERC20PositionWithdrawer.sol (hunk 2 above)
Comment (BrendanChou) [reply contracts/margin/external/ERC20/ERC20PositionWithdrawer.sol]:
```suggestion
```

File: contracts/margin/external/ERC20/ERC20PositionWithdrawer.sol (hunk 1 above)
Comment (BrendanChou) [reply contracts/margin/external/ERC20/ERC20PositionWithdrawer.sol]: Changing back, not necessary to do that call/read if 0 tokens are returned

File: contracts/margin/external/ERC20/ERC20PositionWithdrawer.sol (hunk 2 above)
Comment (BrendanChou) [reply contracts/margin/external/ERC20/ERC20PositionWithdrawer.sol]: Trying out this Suggested Change (beta) thing lol. Removing

File: contracts/margin/external/ERC20/ERC20PositionWithdrawer.sol (hunk 3 above)
Comment (BrendanChou) [reply contracts/margin/external/ERC20/ERC20PositionWithdrawer.sol]: Removing both nonReentrants in general. Only the ERC20Position, the Token, and the ExchangeWrapper/Exchange take over the code execution. Any or all of these are should be well known to the withdrawer and have control of the tokens at some point, so they would be able to "steal" the tokens with or without the ReentrancyGuard

Comment (AntonioJuliano) [reply contracts/margin/external/ERC20/ERC20PositionWithdrawer.sol]: Yeah I agree, I just think we need to be consistent and decide on if we're going to keep adding `nonReentrant` to places where we don't believe it's necessary just to be safe. It's fine to remove these though

**Pull request merged**